Bottom: 056c1c54b51f67c0475b3720b1aec753683bb978
Top:    f079a6c58202017e01841a142e7311cf0b9bee96
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-06-21 14:26:45 +0800

tuntap: don't hold device when peeking packet length

There's no need to hold refcnt of the device during peeking, since
device are loosely coupled with socket. So remove the code for a more
efficient peeking for e.g vhost busy polling.

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 85e14ad..9e6b7b7 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -2461,17 +2461,8 @@ static int tun_ptr_peek_len(void *ptr)
 static int tun_peek_len(struct socket *sock)
 {
 	struct tun_file *tfile = container_of(sock, struct tun_file, socket);
-	struct tun_struct *tun;
-	int ret = 0;
 
-	tun = tun_get(tfile);
-	if (!tun)
-		return 0;
-
-	ret = PTR_RING_PEEK_CALL(&tfile->tx_ring, tun_ptr_peek_len);
-	tun_put(tun);
-
-	return ret;
+	return PTR_RING_PEEK_CALL(&tfile->tx_ring, tun_ptr_peek_len);
 }
 
 /* Ops structure to mimic raw sockets with tun */
